src/operation/watch/mod.rs

```diff
@@ -0,0 +1,104 @@
+#[cfg(test)]
+mod test;
+use super::aggregate::{Aggregate, AggregateTarget};
+use crate::{
+    bson::{doc, Document},
+    change_stream::{ChangeStreamSpecification, ChangeStreamTarget},
+    cmap::{Command, CommandResponse, StreamDescription},
+    error::Result,
+    operation::Operation,
+    options::ChangeStreamOptions,
+};
+#[derive(Debug)]
+pub(crate) struct Watch {
+    target: ChangeStreamTarget,
+    pipeline: Vec<Document>,
+    options: Option<ChangeStreamOptions>,
+    aggregate_operation: Aggregate,
+}
+impl Watch {
+    // Create an aggregate command
+    pub(crate) fn new(
+        target: impl Into<ChangeStreamTarget>,
+        pipeline: impl IntoIterator<Item = Document>,
+        options: Option<ChangeStreamOptions>,
+    ) -> Self {
+        let mut doc: Document = Document::new();
+        match &options {
+            Some(opts) => doc.insert("$changeStream", bson::to_bson(&opts).unwrap()),
+            None => doc.insert("$changeStream", doc! {}),
+        };
+        let mut aggregate_pipeline = Vec::new();
+        let pipeline: Vec<Document> = pipeline.into_iter().collect();
+        let target: ChangeStreamTarget = target.into();
+        match &target {
+            ChangeStreamTarget::Collection(collection) => {
+                let aggregate_target = AggregateTarget::Collection(collection.namespace());
+                aggregate_pipeline.push(doc);
+                aggregate_pipeline.extend(pipeline.clone());
+                let aggregate_operation = Aggregate::new(
+                    aggregate_target,
+                    aggregate_pipeline,
+                    options.clone().map(|opts| opts.get_aggregation_options()),
+                );
+                Self {
+                    target,
+                    pipeline: pipeline.into_iter().collect(),
+                    options,
+                    aggregate_operation,
+                }
+            }
+            ChangeStreamTarget::Database(database) => {
+                let aggregate_target = AggregateTarget::Database(database.name().to_string());
+                aggregate_pipeline.push(doc);
+                aggregate_pipeline.extend(pipeline.clone());
+                let aggregate_operation = Aggregate::new(
+                    aggregate_target,
+                    aggregate_pipeline,
+                    options.clone().map(|opts| opts.get_aggregation_options()),
+                );
+                Self {
+                    target,
+                    pipeline: pipeline.into_iter().collect(),
+                    options,
+                    aggregate_operation,
+                }
+            }
+            ChangeStreamTarget::Cluster(database) => {
+                let aggregate_target = AggregateTarget::Database(database.name().to_string());
+                doc.get_document_mut("$changeStream")
```

Comment:
I think it would be cleaner to do this check before putting the $changeStream document into the pipeline:

```rust
let mut options_doc = Document::new();
append_options(&mut options_doc, &options)?;

if let ChangeStreamTarget::Cluster(..) = target {
	options_doc.insert("allChangesForCluster", true);
}

let mut aggregate_pipeline = vec![options_doc];
aggregate_pipeline.extend(pipeline.clone());
```

Reply:
fixed